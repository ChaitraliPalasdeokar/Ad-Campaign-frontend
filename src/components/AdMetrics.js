// import { ShoppingCartCheckout } from "@mui/icons-material";
// import { Card, Typography } from "@mui/material";

// function ShoppingCart() {
//     return (<div>
        
//         <Typography.Title leve={4}>Dashboard</Typography.Title>
//         {/* <Space direction="horizontal"> */}
//             <DashboardCard title={"Title"} />
//         {/* </Space> */}
//     </div>);
// };

// function DashboardCard({ title }) {
//     return (
//         // <Card>
//         //     {/* <Space direction="horizontal"> */}
//         //         <ShoppingCartCheckout />
//         //         <Statistic title="Orders" value={123888888884} />
//         //     {/* </Space> */}
//         // </Card>
//         <DefaultStatisticsCard
//     );
// };

// export default ShoppingCart;