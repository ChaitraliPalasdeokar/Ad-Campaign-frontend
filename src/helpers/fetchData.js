import axios from "axios";

export const fetchData = async () => {
	try {
        const response = await axios.get(
					"http://localhost:5000/api/v1/data?groupby=source&aggregate=attributed_revenue"
				);
		return response.data;
	} catch (error) {
		console.error(error);
	}
};
