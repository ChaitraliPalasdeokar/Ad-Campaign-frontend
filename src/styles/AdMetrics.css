.ad-metrics {
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
  padding: 20px;
}

.ad-metrics-header {
  font-size: 1.5rem;
  margin-bottom: 10px;
}

.ad-metrics-data {
  display: flex;
  margin-left: 20%;
  margin-right: 20%;
  margin-top: 10px;
  justify-content: space-between;
}

.ad-metrics-item {
  text-align: center;
}

.ad-metrics-label {
  display: block;
  font-size: 1.2rem;
  font-weight: bold;
  margin-bottom: 5px;
}

.ad-metrics-value {
  /* display: block; */
  font-size: 1.5rem;
  /* font-weight: bold; */
}
