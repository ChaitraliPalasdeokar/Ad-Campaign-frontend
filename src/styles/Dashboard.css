.doughnut-chart-container{
  position: relative;
  height: 500px;
  width: 500px;
  padding: auto;
  margin: auto;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
}
.line-chart-container{
  position: relative;
  height: 600px;
  width: 1000px;  
  padding-left: 5%;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
}
.container{
  padding-top: 5%;
  display: flex;
  /* justify-content: space-evenly; */
  flex-wrap: nowrap;
  flex-direction: row;
  justify-content: space-between;
}

.bar-chart-container{
  position: relative;
  height: 1000px; 
  width: 1000px; 
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
}

.clustered-bar-chart-container{
  margin: auto;
  margin-top: 20px;
  height: 400px;
  width:500px;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
  /* margin-left: 10%; */
}