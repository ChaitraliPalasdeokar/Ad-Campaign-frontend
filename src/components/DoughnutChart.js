import React, { useState, useEffect } from "react";
import { Doughnut } from "react-chartjs-2";
import { fetchData } from "../helpers/fetchData";
import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";

const DoughnutChart = (props) => {
	ChartJS.register(ArcElement, Tooltip, Legend);
	const [data, setData] = useState({});

	useEffect(() => {
		const getData = async () => {
			const result = await fetchData(
				"data?groupby=source&aggregate=attributed_conversions"
			);
			setData(result);
		};

		getData();
	}, []);

	const attributed_conversions_object = Object.values(data);
	const attributed_conversions_array = attributed_conversions_object.map((obj) => Object.values(obj)[0]);

	const chartData = {
		labels: Object.keys(data),
		datasets: [
			{
				label: "Attributed Conversions",
				data: attributed_conversions_array,
				backgroundColor: [
					"rgba(255, 99, 132, 0.4)",
					"rgba(54, 162, 235, 0.4)",
					"rgba(255, 206, 86, 0.4)",
					"rgba(75, 192, 192, 0.4)",
					"rgba(153, 102, 255, 0.4)",
					"rgba(255, 159, 64, 0.4)",
					"rgba(120, 140, 20, 0.4)",
					"rgba(150, 225, 11, 0.4)",
					"rgba(11, 250, 11, 0.4)",
					"rgba(55, 34,72, 0.4)",
				],
				borderColor: [
					"rgba(255, 99, 132, 1)",
					"rgba(54, 162, 235, 1)",
					"rgba(255, 206, 86, 1)",
					"rgba(75, 192, 192, 1)",
					"rgba(153, 102, 255, 1)",
					"rgba(255, 159, 64, 1)",
					"rgba(120, 140, 20,1)",
					"rgba(150, 225, 11, 1)",
					"rgba(11, 250, 11, 1)",
					"rgba(55, 34,72, 1)",
				],
				borderWidth: 1,
			},
		],
	};

	const chartOptions = {
		responsive: true,
		maintainAspectRatio: false,
		plugins: {
			legend: {
				display: true,
				position: "right",
			},
			title: {
				display: true,
				text: "Attributed Conversions for each source",
			},
		},
	};

	return <Doughnut {...props} data={chartData} options={chartOptions} />;
};

export default DoughnutChart;
