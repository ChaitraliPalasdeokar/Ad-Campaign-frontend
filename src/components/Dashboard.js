import React, { useState, useEffect } from "react";
import BarChart from "./BarChart";
import ClusteredBarChart from "./ClusteredBarChart";
import { fetchData } from "../helpers/fetchData";
import LineChart from "./LineChart";
import DoughnutChart from "./DoughnutChart";
import AdMetrics from "./Metrics";
import "../styles/Dashboard.css";
const Dashboard = () => {
	const [data, setData] = useState({});
	const [selectedIndex, setSelectedIndex] = useState(-1);
	console.log("In app js", selectedIndex);

	const handleClick = (index) => {
		setSelectedIndex(index);
	};

	useEffect(() => {
		const getData = async () => {
			const result = await fetchData(
				"data?groupby=source&aggregate=attributed_revenue"
			);
			setData(result);
		};

		getData();
	}, []);

	return (
		<div>
			<AdMetrics />
			<div class="page-wrapper">
				<div class="row">
					<div class="wrapper">
						<BarChart data={data} handleClick={handleClick} />
					</div>
					<div class="wrapper">
						<DoughnutChart style={{ height: "100 %", width: "100%" }} />
					</div>
					<div class="wrapper">
						<LineChart
							selectedIndex={selectedIndex}
							style={{ height: "100 %", width: "100%" }}
						/>
					</div>
					<div class="wrapper">
						<ClusteredBarChart />
					</div>
				</div>
			</div>
		</div>
	);
};

export default Dashboard;
