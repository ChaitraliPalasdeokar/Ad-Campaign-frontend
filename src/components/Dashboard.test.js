import React from "react";
import {
	render,
	screen,
	waitFor,
	waitForElementToBeRemoved,
} from "@testing-library/react";
import DoughnutChart from "./DoughnutChart";
import { fetchData } from "../helpers/fetchData";
import AdMetrics from "./Metrics";

import renderer from "react-test-renderer";
import Dashboard from "./Dashboard";
import { act } from "react-dom/test-utils";
// import { create, act } from "react-test-renderer";

jest.mock("react-chartjs-2", () => ({
	Doughnut: () => null,
	Bar: () => null,
	Line: () => null,
}));

jest.mock("../helpers/fetchData", () => ({
	__esModule: true,
	fetchData: jest.fn(),
}));

jest.mock("./LineChart");

describe("DoughnutChart component", () => {
	it("should show dashboard", async () => {
		const mockData = {
			affiliate_prospecting: {
				spends: 114594.37490799998,
			},
			baseline: {
				spends: 10450.631080134954,
			},
		};
		fetchData.mockResolvedValueOnce(mockData);

		const comp = renderer.create(<Dashboard />);

		expect(comp.toJSON()).toMatchSnapshot();
	});
});
