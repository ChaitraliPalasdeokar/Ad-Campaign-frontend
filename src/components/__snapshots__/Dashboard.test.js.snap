// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoughnutChart component should show dashboard 1`] = `
<div>
  <div
    className="ad-metrics"
  >
    <h2
      className="ad-metrics-header"
    >
      Advertising Metrics from June to November, 2022
    </h2>
    <div
      className="ad-metrics-data"
    >
      <div
        className="ad-metrics-item"
      >
        <span
          className="ad-metrics-label"
        >
          Total Revenue:
        </span>
        <span
          className="ad-metrics-value"
        />
      </div>
      <div
        className="ad-metrics-item"
      >
        <span
          className="ad-metrics-label"
        >
          Total Conversions:
        </span>
        <span
          className="ad-metrics-value"
        />
      </div>
      <div
        className="ad-metrics-item"
      >
        <span
          className="ad-metrics-label"
        >
          Total Spend:
        </span>
        <span
          className="ad-metrics-value"
        />
      </div>
    </div>
  </div>
  <div
    class="page-wrapper"
  >
    <div
      class="row"
    >
      <div
        class="column"
      />
      <div
        class="column"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="column"
      />
      <div
        class="column"
      />
    </div>
  </div>
</div>
`;
